src/cript/api/api.py

```diff
@@ -402,6 +406,791 @@ def api_prefix(self):
     def api_version(self):
         return self._api_version
 
+    # ==========================================================
+
+    @staticmethod
+    def remove_keys_from_dict(
+        dictionary,
+        keys_to_remove=[
+            "admin",
+            # "uuid",
+            "updated_by",
+            "created_by",
+            "member",
+            "experiment_count",
+            "inventory_count",
+            "public",
+            "created_at",
+            "email",
+            "model_version",
+            "orcid",
+            "uid",
+            "updated_at",
+            "username",
+            "admin_count",
+            "collection_count",
+            "locked",
+            "material_count",
+            "member_count",
+            "component_count",
+            "computational_forcefield_count",
+            "identifier_count",
+            "computation_count",
+            "component_count",
+            "citation_count",
+            "condition_count",
+            "sample_preparation_count",
+            "property_count",
+        ],
+    ):
+        """Recursively remove keys from a dictionary."""
+        if not isinstance(dictionary, dict):
+            return dictionary
+        for key in keys_to_remove:
+            dictionary.pop(key, None)
+        for key, value in list(dictionary.items()):  # Use list() to avoid RuntimeError during iteration
+            if isinstance(value, dict):
+                API.remove_keys_from_dict(value, keys_to_remove)
+            elif isinstance(value, list):
+                for item in value:
+                    API.remove_keys_from_dict(item, keys_to_remove)
+        return dictionary
+
+    @classmethod
+    def add_to_dict(cls, dictionary, key=None, value=None):
+        # Check if the key exists in the dictionary
+        if key in dictionary:
+            # Check if the value is a list
+            if isinstance(value, list):
+                # Value is a list, extend the existing list with the new one
+                dictionary[key].extend(value)
+            else:
+                # Value is not a list, append it to the list
+                dictionary[key].append(value)
+        else:
+            # Key does not exist, create a new list with the value
+            # If the value is a list, use it directly; otherwise, create a list with one element
+            dictionary[key] = value if isinstance(value, list) else [value]
+
+    # ================================== ADD EXISTING NODES-NAMES===================================
+    # @staticmethod
+    def send_api_patch_existing_nodes_by_name(
+        self,
+        # parent_node: PrimaryBaseNode,
+        parent_node,  # : str,
+        parent_uuid,  #: str,
+        child_class_type,  #: str,
+        existing_child_node_names,  #: list,  # must be list of strings
+    ):
+        """
+        this function will take a list of exact names and add them by uuid
```

Comment:
not really adequate documentation

And I don't really see the need for this to be public.
Users can't be expected to call it.